src/Symfony/Component/Filesystem/Watcher/INotifyWatcher.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Filesystem\Watcher;
+
+use Symfony\Component\Filesystem\Exception\IOException;
+
+/**
+ * @author Pierre du Plessis <pdples@gmail.com>
+ *
+ * @internal
+ */
+final class INotifyWatcher implements WatcherInterface
+{
+    public function watch($path, callable $callback, float $timeout = null): void
+    {
+        $inotifyInit = inotify_init();
+
+        if (false === $inotifyInit) {
+            throw new IOException('Unable initialize inotify.', 0, null, $path);
+        }
+
+        stream_set_blocking($inotifyInit, false);
+
+        $isDir = is_dir($path);
+        $watchers = [];
+
+        if ($isDir) {
+            $watchers[] = inotify_add_watch($inotifyInit, $path, IN_CREATE | IN_DELETE | IN_MODIFY);
+
+            foreach ($this->scanPath("$path/*") as $path) {
```

Comment:
> not only I don't see this mechanism there, but also no inotify_add_watch

`inotify_add_watch` is used [here](https://github.com/flint/Lurker/blob/712d3ef19bef161daa2ba0e0237c6b875587a089/src/Lurker/StateChecker/Inotify/ResourceStateChecker.php#L122)

Checking recursively can maybe be turned off by default and enabled by either a flag or a new method `watchRecursive`

Reply:
> inotify_add_watch is used here

Thanks, GH didn't find it for me.

> Checking recursively can maybe be turned off by default and enabled by either a flag or a new method watchRecursive

I don't mean to have ability to turn recursivity off, I am just trying to figure if there isn't smarter way to go around handling recursivity. First thing we need to clear up is: Is it absolute requirement for inotify PHP extension to create watcher for every subdirectory, instead of just parent directory?